### PR TITLE
Fix perf issue for InstallAnotherVersionAction

### LINKED ISSUE #140525: `Install Another Version...` takes a really long time for the C# extension only
Does this issue occur when all extensions are disabled?: N/A

- VS Code Version: **_1.63.2_**
- OS Version: **_Windows 10 Enterprise, Version 20H2, OS Build 19042.1415_**

I raised [an issue](https://github.com/OmniSharp/omnisharp-vscode/issues/4973) regarding this in **OmniSharp-VSCode**'s repo, but they suggested that the issue should be raised here.

Steps to Reproduce:

For weeks, i've noticed that among the all extensions i've installed, the C# one is the only one has this issue. Go to the Extensions view. Click `Install Another Version...` on the menu for the **C#** extension. Observe that after hanging for a **_long_** time (it took me **56** seconds), the expected list of versions finally shows up.
VSCodeCSharpExtensionInstallAnotherVer

### PR DESCRIPTION
**Pull request by TwitchBronBron**
This PR fixes #140525

There's a significant performance issue with the "Install another version" dropdown in the extensions panel, but it only affects extensions that have undergone an ownership change. For example, [this extension](https://marketplace.visualstudio.com/items?itemName=RokuCommunity.brightscript) that I maintain used to be published under the publisher name "celsoaf", but is now published under the publisher name "rokucommunity". (I worked with vsmarketplace team to migrate the extension to a new publisher name without losing all our stats and information). 

The issue is caused by certain artifacts on the server no longer existing (or using the wrong URLs, not sure which), resulting in very slow (300-500ms) 403 and 404 http responses from the server. This drastically slows down the process of gathering all version information since that process is run sequentially [here](https://github.com/microsoft/vscode/blob/bd25c294cd7d5d0fbc347be77a76d9aef0685b5c/src/vs/platform/extensionManagement/common/extensionGalleryService.ts#L842). Ideally the vsmarketplace team could speed up their 403 and 404 error response times or include additional version metadata so vscode doesn't need to run an HTTP query for every version. However, I have created a fix that seems to work around the issue.

To mitigate the performance issues, rather than running the queries one at a time, we instead run them all at the same time. Consider the results:

Before my fix: 57 seconds
After my fix: 1 second

vscode

### CODE REVIEW

**Review by iObject — APPROVED** [review]

File: src/vs/platform/extensionManagement/common/extensionGalleryService.ts [hunk 1]
```diff
@@ -882,15 +882,22 @@ abstract class AbstractExtensionGalleryService implements IExtensionGalleryServi
...
-			try {
-				if (result[result.length - 1]?.version !== version.version && await this.isRawExtensionVersionCompatible(version, includePreRelease, allTargetPlatforms, targetPlatform)) {
-					result.push({ version: version.version, date: version.lastUpdated, isPreReleaseVersion: isPreReleaseVersion(version) });
-				}
-			} catch (error) { /* Ignore error and skip version */ }
-		}
-		return result;
+		const seenVersions = new Set<string>();
+		// Run checks in parallel to reduce wait time (fixes slow performance related to extensions with ownership changes)
+		const result = await Promise.all(
```
Comment (axetroy) [inline src/vs/platform/extensionManagement/common/extensionGalleryService.ts:887]: How about use [p-map](https://www.npmjs.com/package/p-map) for control concurrently?

Comment (TwitchBronBron) [reply src/vs/platform/extensionManagement/common/extensionGalleryService.ts:887]: I'm a new contributor to vscode. What's vscode's policy on adding more npm packages? I'd be happy to add this if it's a normal thing to do, as we can help manage how many will run at the same time.

**Comment by TwitchBronBron** [comment]
@sandy081 is there anything I can do to help keep this moving?

**Comment by sandy081** [comment]
I will take a look at it and get back to you. Thanks.

**Review by sandy081 — APPROVED** [review]

**Pull request merged**